Modules/_testinternalcapi.c

```diff
@@ -35,6 +36,38 @@ get_recursion_depth(PyObject *self, PyObject *Py_UNUSED(args))
 }
 
 
+static PyObject*
+codecs_unregister(PyObject *self, PyObject *args)
+{
+    PyObject *search_function;
+
+    if (!PyArg_ParseTuple(args, "O:codecs_unregister",
+                          &search_function)) {
+        return NULL;
+    }
+    if (search_function == NULL) {
+        return NULL;
+    }
+
+    PyInterpreterState *interp = PyInterpreterState_Get();
+    if (interp->codec_search_path == NULL) {
+        return NULL;
+    }
+
+    Py_ssize_t n = PyList_Size(interp->codec_search_path);
+    PyObject *list = PyList_New(0);
+    for (Py_ssize_t i = 0; i < n; i++) {
+        PyObject *item = PyList_GetItem(interp->codec_search_path, i);
+        if (item != search_function) {
+            PyList_Append(list, item);
+        }
+    }
+    Py_SETREF(interp->codec_search_path, list);
+
+    Py_RETURN_NONE;
```

Comment:
Would not be easier to use `PyObject_CallMethod()`.
```suggestion
    return PyObject_CallMethod(interp->codec_search_path, "remove", "O",
                               search_function);
```

Reply:
Hi, serhiy. The latest PR have updated in https://github.com/python/cpython/pull/22360/files.
victor recommend use `PyList_SetSlice` in here.
IMHO, compare to `PyList_SetSlice`, `PyList_Remove` would be more exact.
You give me another option. I am not sure which one is the best idea, Lol~

Reply:
`PyList_SetSlice()` allows to remove an item by index. `list.remove()` removes it by value.